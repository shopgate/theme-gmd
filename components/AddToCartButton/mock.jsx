/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * Mocked AddToCartButton.
 * @type {MockedAddToCartButton}
 */
// eslint-disable-next-line react/prefer-stateless-function
export const MockedAddToCartButton = class extends Component {
  static propTypes = {
    handleAddToCart: PropTypes.func,
  };
  static defaultProps = {
    handleAddToCart: () => {},
  };
  /**
   * Renders mocked button.
   * @return {JSX}
   */
  render() {
    return (
      <button onClick={this.props.handleAddToCart} />
    );
  }
};
