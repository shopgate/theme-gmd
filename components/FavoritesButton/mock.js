/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

const mockedProduct1 = {
  productId: 'foo',
};

const currentProduct = {
  productId: 'bar',
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateNotOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId],
    },
  },
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
      [currentProduct.productId]: {
        productData: currentProduct,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId, currentProduct.productId],
    },
  },
};

/**
 * Mocked state without data.
 * @type {{product: {currentProduct: {}, favorites: {}}}}
 */
export const mockedStateEmpty = {
  product: {
    currentProduct: {},
    productsById: {},
  },
  favorites: {
    products: {
      ids: [],
    },
  },
};
