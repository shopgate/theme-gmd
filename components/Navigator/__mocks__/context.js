import React from 'react';

export const defaultContext = {
  searchField: false,
  toggleSearchField: jest.fn(),
  searchQuery: 'shirt',
  setSearchQuery: jest.fn(),
};
let context;

export const NavigatorContext = ({
  Provider(props) {
    /* eslint-disable react/prop-types */
    context = props.value || defaultContext;
    return React.createElement('div', props, props.children);
    /* eslint-enable react/prop-types */
  },
  Consumer(props) {
    return props.children(context || defaultContext);
  },
});
