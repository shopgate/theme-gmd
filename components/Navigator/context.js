import React from 'react';

export const NavigatorContext = React.createContext();
