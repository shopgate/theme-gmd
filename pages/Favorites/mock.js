/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */
const foo = {
  productData: {
    name: 'Foo',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'foo',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

const bar = {
  productData: {
    name: 'Bar',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'bar',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

export const mockedEmptyState = {
  product: {
    productsById: {
      foo,
    },
  },
  favorites: {
    products: {
      ids: [],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: ['foo'],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedNotReadyState = {
  ...mockedEmptyState,
  favorites: {
    products: {},
    isFetching: false,
  },
};

export const mockedNextProps = {
  products: [
    foo.productData,
    bar.productData,
  ],
};

