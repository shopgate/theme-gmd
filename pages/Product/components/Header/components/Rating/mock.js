/*
 *  Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

/**
 * Mock for getElementById
 * @param {function} scrollSpy Element.scroll spy function.
 * @returns {function}
 */
const getElementById = scrollSpy => () => ({
  offsetTop: 100,
  closest() {
    return {
      scroll: scrollSpy,
    };
  },
});

export {
  getElementById,
};
