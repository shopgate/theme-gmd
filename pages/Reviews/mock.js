/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  mockedStateWithoutReview,
} from 'Components/Reviews/mock';

export const mockedState = {
  ...mockedStateWithoutReview,
  reviews: {
    reviewsByHash: {},
  },
  history: {
    state: {
      title: '',
      viewTop: true,
    },
  },
};
