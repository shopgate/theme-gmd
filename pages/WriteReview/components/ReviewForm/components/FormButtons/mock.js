/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked state with product only. Reviews not fetched.
 * @type {Object}
 */
export const mockedState = {
  history: {
    pathname: '/item/666f6f/write_review',
  },
  view: {
    isLoading: false,
  },
};
