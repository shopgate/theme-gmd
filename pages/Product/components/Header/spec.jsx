import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import { defaultContext } from './../../__mocks__/context';
import { basicProductState } from './../mock';

const mockedStore = configureStore([thunk]);

jest.mock('Components/Reviews/components/Header', () => () => <div />);

jest.mock('./../../context');

describe.skip('<ProductHeader>', () => {
  /**
   * @param {Object} state A redux store.
   * @returns {JSX}
   */
  const createComponent = (state) => {
    // eslint-disable-next-line global-require
    const ProductHeader = require('./index').default;
    const store = mockedStore(state);
    return mount(
      <Provider store={store}>
        <ProductHeader productId={defaultContext.productId} />
      </Provider>,
      mockRenderOptions
    );
  };

  const components = [
    'CTAButtons',
    'Manufacturer',
    'Availability',
    'Rating',
    'Name',
    'Price',
    'PriceStriked',
    'Shipping',
    'Tiers',
  ];

  beforeEach(() => {
    jest.resetModules();
  });

  it('should render', () => {
    const { productId } = basicProductState.product.currentProduct;
    defaultContext.productId = productId;
    const cmp = createComponent(basicProductState);
    expect(cmp).toMatchSnapshot();
    components.forEach((c) => {
      expect(cmp.find(c).exists()).toBe(true);
    });
  });
});
