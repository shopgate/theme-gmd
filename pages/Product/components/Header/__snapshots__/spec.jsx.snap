// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductHeader> should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <pure(ProductHeader)
    productId="913"
  >
    <ProductHeader
      productId="913"
    >
      <Consumer>
        <div
          className="css-he61ou"
        >
          <Portal
            name="product.ctas.before"
            props={null}
          />
          <Portal
            name="product.ctas"
            props={null}
          >
            <Connect(CTAButtons)
              productId={null}
            >
              <CTAButtons
                dispatch={[Function]}
                isFavorite={false}
                productId={null}
              >
                <div
                  className="css-7azkrx"
                >
                  <Connect(FavoritesButton)
                    active={false}
                    className="css-13l587d"
                    productId={null}
                    rippleClassName="css-1tdptiz"
                  >
                    <FavoritesButton
                      active={false}
                      addFavorites={[Function]}
                      className="css-13l587d"
                      noShadow={false}
                      onRippleComplete={[Function]}
                      once={false}
                      productId={null}
                      removeFavorites={[Function]}
                      removeThrottle={0}
                      removeWithRelatives={false}
                      rippleClassName="css-1tdptiz"
                    />
                  </Connect(FavoritesButton)>
                  <Component
                    productId={null}
                  >
                    <Consumer>
                      <Connect(CartButton)
                        conditioner={
                          Conditioner {
                            "conditioners": Map {},
                          }
                        }
                        options={Object {}}
                        productId={null}
                      >
                        <CartButton
                          addToCart={[Function]}
                          conditioner={
                            Conditioner {
                              "conditioners": Map {},
                            }
                          }
                          disabled={true}
                          loading={true}
                          options={Object {}}
                          productId={null}
                        >
                          <FloatingActionButton
                            background="#fa5400"
                            className="css-mp6sl7"
                            onClick={[Function]}
                            raised={true}
                            ref={null}
                            size="big"
                            type="button"
                          >
                            <button
                              className="css-1dbx872 css-mp6sl7 css-1kx79ob css-1g5n8t6"
                              onClick={[Function]}
                              style={
                                Object {
                                  "background": "#fa5400",
                                }
                              }
                              type="button"
                            >
                              <IndicatorCircle
                                color="#fff"
                                paused={false}
                                size={32}
                                strokeWidth={4}
                              >
                                <svg
                                  className="css-12jxe4c"
                                  height={32}
                                  viewBox="25 25 50 50"
                                  width={32}
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <circle
                                    className="css-1srioz5"
                                    cx="50"
                                    cy="50"
                                    r="20"
                                  />
                                </svg>
                              </IndicatorCircle>
                            </button>
                          </FloatingActionButton>
                        </CartButton>
                      </Connect(CartButton)>
                    </Consumer>
                  </Component>
                </div>
              </CTAButtons>
            </Connect(CTAButtons)>
          </Portal>
          <Portal
            name="product.ctas.after"
            props={null}
          />
          <Portal
            name="product.rating.before"
            props={null}
          />
          <Portal
            name="product.rating"
            props={null}
          >
            <Connect(Rating)
              productId={null}
            >
              <Rating
                dispatch={[Function]}
                productId={null}
                rating={null}
              />
            </Connect(Rating)>
          </Portal>
          <Portal
            name="product.rating.after"
            props={null}
          />
          <Portal
            name="product.name.before"
            props={null}
          />
          <Portal
            name="product.name"
            props={null}
          >
            <Connect(Name)
              productId={null}
            >
              <Name
                dispatch={[Function]}
                name={null}
                productId={null}
              >
                <div
                  className="css-19910ke"
                >
                  <PlaceholderLabel
                    className="css-3bon8p"
                    ready={false}
                    style={null}
                  >
                    <div
                      className="css-qd8qpa css-3bon8p"
                      style={null}
                    />
                  </PlaceholderLabel>
                </div>
              </Name>
            </Connect(Name)>
          </Portal>
          <Portal
            name="product.name.after"
            props={null}
          />
          <Portal
            name="product.info.before"
            props={null}
          />
          <Grid
            className=""
            component="div"
            wrap={false}
          >
            <div
              className=" css-h7r55a"
            >
              <Portal
                name="product.info"
                props={null}
              >
                <GridItem
                  className=""
                  component="div"
                  grow={1}
                  shrink={1}
                >
                  <div
                    className=" css-1gs0ko2"
                  >
                    <Portal
                      name="product.info.row1"
                      props={null}
                    >
                      <div
                        className="css-qboqsd"
                      >
                        <Portal
                          name="product.manufacturer.before"
                          props={null}
                        />
                        <Portal
                          name="product.manufacturer"
                          props={null}
                        >
                          <Connect(Manufacturer)
                            productId={null}
                          >
                            <Manufacturer
                              dispatch={[Function]}
                              manufacturer={null}
                              productId={null}
                            >
                              <div
                                className="css-s3s3r0"
                                data-test-id="manufacturer: null"
                              >
                                <PlaceholderLabel
                                  className="css-1n196ny"
                                  ready={false}
                                  style={null}
                                >
                                  <div
                                    className="css-qd8qpa css-1n196ny"
                                    style={null}
                                  />
                                </PlaceholderLabel>
                              </div>
                            </Manufacturer>
                          </Connect(Manufacturer)>
                        </Portal>
                        <Portal
                          name="product.manufacturer.after"
                          props={null}
                        />
                      </div>
                      <div
                        className="css-qboqsd"
                      >
                        <Portal
                          name="product.shipping.before"
                          props={null}
                        />
                        <Portal
                          name="product.shipping"
                          props={null}
                        >
                          <Connect(Shipping)
                            productId={null}
                          >
                            <Shipping
                              dispatch={[Function]}
                              productId={null}
                              shipping={null}
                            >
                              <PlaceholderLabel
                                className="css-1n196ny"
                                ready={false}
                                style={null}
                              >
                                <div
                                  className="css-qd8qpa css-1n196ny"
                                  style={null}
                                />
                              </PlaceholderLabel>
                            </Shipping>
                          </Connect(Shipping)>
                        </Portal>
                        <Portal
                          name="product.shipping.after"
                          props={null}
                        />
                      </div>
                      <div
                        className="css-qboqsd"
                      >
                        <Portal
                          name="product.availability.before"
                          props={null}
                        />
                        <Portal
                          name="product.availability"
                          props={null}
                        >
                          <Connect(Availability)
                            productId={null}
                          >
                            <Availability
                              availability={null}
                              dispatch={[Function]}
                              productId={null}
                            >
                              <PlaceholderLabel
                                className="css-1n196ny"
                                ready={false}
                                style={null}
                              >
                                <div
                                  className="css-qd8qpa css-1n196ny"
                                  style={null}
                                />
                              </PlaceholderLabel>
                            </Availability>
                          </Connect(Availability)>
                        </Portal>
                        <Portal
                          name="product.availability.after"
                          props={null}
                        />
                      </div>
                    </Portal>
                  </div>
                </GridItem>
                <GridItem
                  className="css-1i9coa0"
                  component="div"
                  grow={0}
                  shrink={1}
                >
                  <div
                    className="css-1i9coa0"
                  >
                    <Portal
                      name="product.info.row2"
                      props={null}
                    >
                      <div
                        className="css-1l9nix6"
                      >
                        <Portal
                          name="product.price-striked.before"
                          props={null}
                        />
                        <Portal
                          name="product.price-striked"
                          props={null}
                        >
                          <Connect(PriceStriked)
                            options={Object {}}
                            productId={null}
                          >
                            <PriceStriked
                              dispatch={[Function]}
                              options={Object {}}
                              price={null}
                              productId={null}
                            >
                              <PlaceholderLabel
                                className="css-1n196ny"
                                ready={false}
                                style={null}
                              >
                                <div
                                  className="css-qd8qpa css-1n196ny"
                                  style={null}
                                />
                              </PlaceholderLabel>
                            </PriceStriked>
                          </Connect(PriceStriked)>
                        </Portal>
                        <Portal
                          name="product.price-striked.after"
                          props={null}
                        />
                      </div>
                      <div
                        className="css-1l9nix6"
                      >
                        <Portal
                          name="product.price.before"
                          props={null}
                        />
                        <Portal
                          name="product.price"
                          props={null}
                        >
                          <Connect(Price)
                            options={Object {}}
                            productId={null}
                          >
                            <Price
                              dispatch={[Function]}
                              options={Object {}}
                              price={null}
                              productId={null}
                            >
                              <PlaceholderLabel
                                className="css-q720h0"
                                ready={false}
                                style={null}
                              >
                                <div
                                  className="css-qd8qpa css-q720h0"
                                  style={null}
                                />
                              </PlaceholderLabel>
                            </Price>
                          </Connect(Price)>
                        </Portal>
                        <Portal
                          name="product.price.after"
                          props={null}
                        />
                      </div>
                      <div
                        className="css-1l9nix6"
                      >
                        <Portal
                          name="product.price-info.before"
                          props={null}
                        />
                        <Portal
                          name="product.price-info"
                          props={null}
                        >
                          <Connect(PriceInfo)
                            options={Object {}}
                            productId={null}
                          >
                            <PriceInfo
                              dispatch={[Function]}
                              options={Object {}}
                              price={null}
                              productId={null}
                            >
                              <PlaceholderLabel
                                className="css-q720h0"
                                ready={false}
                                style={null}
                              >
                                <div
                                  className="css-qd8qpa css-q720h0"
                                  style={null}
                                />
                              </PlaceholderLabel>
                            </PriceInfo>
                          </Connect(PriceInfo)>
                        </Portal>
                        <Portal
                          name="product.price-info.after"
                          props={null}
                        />
                      </div>
                      <div
                        className="css-1l9nix6"
                      >
                        <Portal
                          name="product.tiers.before"
                          props={null}
                        />
                        <Portal
                          name="product.tiers"
                          props={null}
                        >
                          <Connect(Tiers)
                            options={Object {}}
                            productId={null}
                          >
                            <Tiers
                              dispatch={[Function]}
                              options={Object {}}
                              price={null}
                              productId={null}
                            />
                          </Connect(Tiers)>
                        </Portal>
                        <Portal
                          name="product.tiers.after"
                          props={null}
                        />
                      </div>
                    </Portal>
                  </div>
                </GridItem>
              </Portal>
            </div>
          </Grid>
          <Portal
            name="product.info.after"
            props={null}
          />
        </div>
      </Consumer>
    </ProductHeader>
  </pure(ProductHeader)>
</Provider>
`;
